src/org/thoughtcrime/securesms/ConversationActivity.java

```diff
@@ -743,6 +758,49 @@ private void handleViewPinnedMessages() {
     startActivity(intent);
   }
 
+  private void handleSearch(MenuItem item) {
+    bottomPanel = (InputPanel) findViewById(R.id.bottom_panel);
+    searchView = (SearchView)  findViewById(R.id.custom_search);
+
+    searchView.setOnQueryTextListener(new SearchInitiatedListener());
+    upArrow.setOnClickListener(new NextSearchResultListener());
+    downArrow.setOnClickListener(new PreviousSearchResultListener());
+
+    if(!this.isSearchMode) {
+      showSearchMode(bottomPanel, searchView);
+    } else {
+      hideSearchMode();
+    }
+    this.isSearchMode = !isSearchMode;
```

Comment:
`hideSearchMode` already does this ... is this a bug?

Reply:
assuming because the showSearchMode doesnt do it above it, so it just sets it to the opposite of what it what when it reaches there

Reply:
I think we can move this.isSearchMode = !isSearchMode; inside the "if" body @shelbaz , I will test and see if it has any implications

Reply:
hideSearchMode already changes the variable, why is being flipped again?